src/main/java/org/fenixedu/academic/ui/rest/PhotographController.java

```diff
@@ -0,0 +1,145 @@
+
+@Path("/user/photo")
+public class PhotographController {
+
+    public static int MAX_PHOTO_SIZE = 1048576; //1M
+    private Gson gson = new GsonBuilder().create();
+
+    @GET
+    @Path("/{username:.+}")
+    @Produces("image/*")
+    public Response get(@PathParam("username") String username, @QueryParam(value = "s") Integer size,
+            @HeaderParam(value = "If-None-Match") String ifNoneMatch) throws IOException {
+
+        if (size == null || size <= 0) {
+            size = 100;
+        }
+        if (size > 512) {
+            size = 512;
+        }
+
+        User user = User.findByUsername(username);
+
+        if (user != null && user.getPerson() != null) {
+            final Photograph personalPhoto =
+                    user.getPerson().isPhotoAvailableToCurrentUser() ? user.getPerson().getPersonalPhoto() : null;
+
+            EntityTag entityTag =
+                    new EntityTag(personalPhoto == null ? "mm-av" : personalPhoto.getExternalId() + "-" + size, true);
+            if (!StringUtils.isBlank(ifNoneMatch)
+                    && entityTag.getValue().equals(ifNoneMatch.substring(3, ifNoneMatch.length() - 1))) {
+                return Response.status(Response.Status.NOT_MODIFIED).build();
+            }
+
+            CacheControl cacheControl = new CacheControl();
+            cacheControl.setMaxAge(1209600);
+
+            // ChronoUnit.WEEKS was not supported so I used 14 days
+            Response.ResponseBuilder responseBuilder = Response.status(Response.Status.OK)
+                    .expires(Date.from(Instant.now().plus(14, ChronoUnit.DAYS))).cacheControl(cacheControl).tag(entityTag);
+            if (personalPhoto != null) {
+                responseBuilder.type(personalPhoto.getOriginal().getPictureFileFormat().getMimeType());
+                responseBuilder.entity(personalPhoto.getCustomAvatar(size, size, PictureMode.ZOOM));
+                return responseBuilder.build();
+            } else {
+                try (InputStream mm = PhotographController.class.getClassLoader()
+                        .getResourceAsStream("META-INF/resources/img/mysteryman.png")) {
+                    responseBuilder.type("image/png");
+                    responseBuilder.entity(Avatar.process(mm, "image/png", size));
+                    return responseBuilder.build();
+                }
+            }
+        }
+
+        throw BennuCoreDomainException.resourceNotFound(username);
+    }
+
+    @GET
+    @Path(value = "{size}/{username:.+}")
+    @Produces("image/*")
+    public Response getWithSize(@PathParam("username") String username, @PathParam("size") Integer size,
+            @HeaderParam(value = "If-None-Match") String ifNoneMatch) throws IOException {
+        return get(username, size, ifNoneMatch);
+    }
+
+    @POST
+    @Path(value = "/upload")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response upload(final PhotographForm jsonPhotographForm) {
```

Comment:
@joaofreitas-qubit I would say this upload endpoint is no longer used since the upload should be done via a uilayer component. I'm still approving this PR but might add some extra information to this later.

Reply:
Just validaded the upload method is no longer needed.